src/main/java/com/karankumar/bookproject/backend/repository/BookRepository.java

```diff
@@ -19,9 +19,13 @@
 
 import com.karankumar.bookproject.backend.entity.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 import java.util.List;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
     List<Book> findByTitleContainingIgnoreCase(String title);
+
+    @Query("SELECT b FROM Book b LEFT JOIN FETCH b.author LEFT JOIN FETCH  b.predefinedShelf")
+    List<Book> findAllBooksFetchEagerly();
```

Comment:
How come you're fetching the books eagerly?

Reply:
Since existings tests failed when fetching lazily i added a query for eagerly fetching. 
If eagerly fetching is not necessary functionality i can modify existing tests to be compatible with lazy fetching

Reply:
@andrewass Yes, could you please change the tests to work with lazy fetching, please? Thanks!